src/main/java/org/springframework/data/couchbase/core/query/OptionsBuilder.java

```diff
@@ -67,21 +73,21 @@ public class OptionsBuilder {
 	static QueryOptions buildQueryOptions(Query query, QueryOptions options, QueryScanConsistency scanConsistency) {
 		options = options != null ? options : QueryOptions.queryOptions();
 		if (query.getParameters() != null) {
-			if (query.getParameters() instanceof JsonArray) {
+			if (query.getParameters() instanceof JsonArray && !((JsonArray) query.getParameters()).isEmpty()) {
 				options.parameters((JsonArray) query.getParameters());
-			} else {
+			} else if( query.getParameters() instanceof JsonObject && !((JsonObject)query.getParameters()).isEmpty()){
```

Comment:
NIT: formatting

IntelliJ tip: Command-Option-Shift-L, then select "Only changes uncommitted to VCS"